fuel-vm/src/storage/memory.rs

```diff
@@ -204,30 +203,28 @@ impl StorageRead<ContractsRawCode> for MemoryStorage {
         &self,
         key: &ContractId,
         buf: &mut [u8],
-    ) -> Result<Option<usize>, Self::Error> {
-        Ok(self
-            .memory
-            .contracts
-            .get(key)
-            .and_then(|c| c.as_ref().read(buf).ok()))
+    ) -> Result<Option<usize>, StorageError> {
+        Ok(self.memory.contracts.get(key).map(|c| {
+            let len = buf.len().min(c.as_ref().len());
+            buf.copy_from_slice(&c.as_ref()[..len]);
+            buf.len()
+        }))
```

Comment:
Hmm, it seems not super related to this change=)

Do we really want to copy into the buffer silently if the sizes are different?